src/client.tsx

```diff
@@ -2,15 +2,32 @@
 
 import { render } from "hono/jsx/dom";
 import { Field } from "./game";
+import { hc } from "hono/client";
+import { AppType } from ".";
 
 const GRID_SIZE = 20;
 const field = new Field();
+const { api } = hc<AppType>(import.meta.env.BASE_URL);
+const res = await api.field.$get();
+
+let values: number[];
+
+if (res.ok) {
+	const data: { result: string } = await res.json();
+	const uint8Array = new Uint8Array(data.result.length);
+
+	for (let i = 0; i < data.result.length; i++) {
+		uint8Array[i] = data.result.charCodeAt(i);
+	}
+
+	values = Array.from(uint8Array);
+}
```

Comment:
all of this can happen on the backend, reduce the computation on the client and improve readability a bit here

```ts
// path: src/api.ts

  .get("/field", async (c) => {
    const redis = c.get("redis");
    const field = await redis.get<string>("field");
    if (!field) return c.notFound();
    return c.json({
      result: Array.from(new Uint8Array(field.length), (_, index) =>
        field.charCodeAt(index),
      ),
    });
  })
```